openerp/models.py

```diff
@@ -955,40 +954,43 @@ def get_values_clause_bulk(lst):
         to_create = []
         for res_id, name in new_names.items():
             to_create += [{
-                'model': first_item._name,
+                'model': self._name,
                 'res_id': res_id,
                 'module': '__export__',
                 'name': name
             }]
 
-            xml_id_cache[first_item._name][res_id] = '__export__.' + name
+            xml_id_cache[self._name][res_id] = '__export__.' + name
 
         # Do a bulk insert in the database for new records
         if to_create:
             clause, values = get_values_clause_bulk(to_create)
             self.env.cr.execute("INSERT INTO ir_model_data %s" % (clause), values)
 
     @api.multi
-    def __export_rows(self, fields, all_records=False):
```

Comment:
I now pass xml_id_cache recursively so that I don't start with an empty cache when exporting relational fields.